espnet/bin/asr_train.py

```diff
@@ -32,6 +33,10 @@ def get_parser():
 
     parser.add_argument('--ngpu', default=None, type=int,
                         help='Number of GPUs. If not given, use all visible devices')
+    parser.add_argument('--train-dtype', default="float32",
+                        choices=["float16", "float32", "float64", "O0", "O1", "O2", "O3"],
+                        help='Data type for training (only pytorch backend). '
+                        'O0,O1,.. flags require apex. apex automatically selects float precision.')
```

Comment:
I think it is better to add explanation of each option or add url of apex documentation.
Another option is to rename 01 to something like float16-float32-mixed.
How do you think?

Reply:
I will add url to apex.amp doc. https://nvidia.github.io/apex/amp.html#opt-levels